test/multidomain.jl

```diff
@@ -0,0 +1,359 @@
+
+    dx, dy = pixelsizes(imgdomain)
+    U_vals = fftshift(fftfreq(500, 1 / dx))
+    V_vals = fftshift(fftfreq(500, 1 / dy))
+
+    # Repeat U and V to match Fr dimensions
+    U_repeated = repeat(vec(U_vals); outer=(length(Fr)))
+    V_repeated = repeat(vec(V_vals); outer=(length(Fr)))
+    Fr_repeated = repeat(Fr; inner=(Int(length(vec(U_vals)))))
+
+    # Repeat U and V and Fr to match Ti dimensions
+    U_repeated = repeat(U_repeated; outer=(length(Ti)))
+    V_repeated = repeat(V_repeated; outer=(length(Ti)))
+    Fr_repeated = repeat(Fr_repeated; outer=(length(Ti)))
+    Ti_repeated = repeat(Ti; inner=(Int(length(U_repeated) / length(Ti))))
+
+    visdomain = UnstructuredDomain((; U=U_repeated, V=V_repeated, Fr=Fr_repeated,
+                                    Ti=Ti_repeated))
+    p = FourierDualDomain(imgdomain, visdomain, NFFTAlg())
+
+    return p
+end
+
+function domains3d_time(Nx, Nt)
+    X = Y = range(-12, 12; length=Nx)
+    Ti = sort(10 * rand(Nt))
+
+    imgdomain = RectiGrid((; X, Y, Ti))
+
+    dx, dy = pixelsizes(imgdomain)
+    U_vals = fftshift(fftfreq(500, 1 / dx))
+    V_vals = fftshift(fftfreq(500, 1 / dy))
+
+    # Repeat U and V to match Fr dimensions
+    U_repeated = repeat(vec(U_vals); outer=(length(Ti)))
+    V_repeated = repeat(vec(V_vals); outer=(length(Ti)))
+    Ti_repeated = repeat(Ti; inner=(Int(length(vec(U_vals)))))
+
+    visdomain = UnstructuredDomain((; U=U_repeated, V=V_repeated, Ti=Ti_repeated))
+    p = FourierDualDomain(imgdomain, visdomain, NFFTAlg())
+
+    return p
+end
+
+function domains3d_freq(Nx, Nf)
+    X = Y = range(-12, 12; length=Nx)
+    Fr = sort(1e11 * rand(Nf))
+
+    imgdomain = RectiGrid((; X, Y, Fr))
+
+    dx, dy = pixelsizes(imgdomain)
+    U_vals = fftshift(fftfreq(500, 1 / dx))
+    V_vals = fftshift(fftfreq(500, 1 / dy))
+
+    # Repeat U and V to match Fr dimensions
+    U_repeated = repeat(vec(U_vals); outer=(length(Fr)))
+    V_repeated = repeat(vec(V_vals); outer=(length(Fr)))
+    Fr_repeated = repeat(Fr; inner=(Int(length(vec(U_vals)))))
+
+    visdomain = UnstructuredDomain((; U=U_repeated, V=V_repeated, Fr=Fr_repeated))
+    p = FourierDualDomain(imgdomain, visdomain, NFFTAlg())
+
+    return p
+end
+
+function domain2d(Nx)
+    X = Y = range(-12, 12; length=Nx)
+
+    imgdomain = RectiGrid((; X, Y))
+
+    dx, dy = pixelsizes(imgdomain)
+    U_vals = fftshift(fftfreq(500, 1 / dx))
+    V_vals = fftshift(fftfreq(500, 1 / dy))
+
+    visdomain = UnstructuredDomain((; U=vec(U_vals), V=vec(V_vals)))
+    p = FourierDualDomain(imgdomain, visdomain, NFFTAlg())
+
+    return p
+end
+
+# Function to calculate visibilities
+function foo4D(x, p)
+    cimg = ContinuousImage(IntensityMap(x, VLBISkyModels.imgdomain(p)), BSplinePulse{3}())
+    vis = VLBISkyModels.visibilitymap_numeric(cimg, p)
+    return sum(abs2, vis)
+end
+
+# Test function to check autodiff
+function check4dautodiff(Nx, Nt, Nf, x, dx)
+    p = domains4d(Nx, Nt, Nf)
+    Enzyme.API.runtimeActivity!(true)
```

Comment:
You don't need to set runtimeActivity! everytime. This can just be put at the top of the script.